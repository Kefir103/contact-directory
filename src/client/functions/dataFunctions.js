import { orderBy } from 'lodash';

export const getFilteredElements = (elements, filterText, filterFields) => {
    let filteredElements = [...elements];
    filterFields.forEach(
        (field) =>
            (filteredElements = filteredElements.filter((element) =>
                element[field].toString().toLowerCase().includes(filterText.toLowerCase())
            ))
    );
    return filteredElements;
};

export const getSortedElements = (elements, sortingMap) => {
    return orderBy(elements, Array.from(sortingMap.keys()), Array.from(sortingMap.values()));
};

export const getElementsByPage = (elements, page) => {
    const elementsLengthByPage = page * 50;
    const startElementsLengthByPage = (page - 1) * 50;
    const currentElementsArray = [];
    for (let i = startElementsLengthByPage; i < elementsLengthByPage && i < elements.length; ++i) {
        currentElementsArray.push(elements[i]);
    }
    return currentElementsArray;
};

export const getPagesCount = (elements) => {
    return Math.ceil(elements.length / 50);
};

export const getAppElements = (fullData, filter, sortingMap) => {
    let currentElements = [...fullData];
    if (filter) {
        currentElements = getFilteredElements(
            currentElements,
            filter.filterText,
            filter.filterFields
        );
    }
    if (sortingMap.size) {
        currentElements = getSortedElements(currentElements, sortingMap);
    }
    return currentElements;
};
