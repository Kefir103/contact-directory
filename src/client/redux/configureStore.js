import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/index';
import { rootReducer } from './reducers/rootReducer';
import thunk from 'redux-thunk';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
