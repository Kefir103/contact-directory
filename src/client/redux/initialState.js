export const initialState = {
    filter: {
        sortingMap: new Map([]),
        filterText: '',
        filterFields: [],
        currentPage: 1,
    },
    data: {
        currentElement: {
            elementInfo: undefined,
            elementIndex: -1,
        },
        elements: [],
        currentElements: [],
        filteredElements: undefined,
        inputContainer: {
            isAddButtonDisabled: false,
            isFormOpen: false,
            inputElement: {
                id: 0,
                firstName: '',
                lastName: '',
                email: '',
                phone: '',
            },
            validInputs: {
                id: false,
                firstName: false,
                lastName: false,
                email: false,
                phone: false,
            },
            addingStatus: {
                addingStatusText: '',
                isError: false,
            },
        },
    },
    appStatus: {
        isLoading: false,
        error: undefined,
    },
};
