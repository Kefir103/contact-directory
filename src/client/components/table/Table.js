import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loadData, setSelectedElement } from '../../redux/actions/dataActions';
import { setSortingMap } from '../../redux/actions/filterActions';
import TablePanel from './TablePanel';
import TableElement from './TableElement';
import Loader from '../Loader';

const Table = (props) => {
    const renderTable = (elements, filteredElements) => {
        if (props.isLoading === false) {
            if (filteredElements && filteredElements.length) {
                return [
                    <TablePanel />,
                    ...filteredElements.map((element) => <TableElement element={element} />),
                ];
            } else if (filteredElements && !filteredElements.length) {
                return (
                    <p id={'filter-error-message'}>
                        Нет элементов списка, которые соответствуют Вашему запросу
                    </p>
                );
            } else if (elements.length) {
                return [
                    <TablePanel />,
                    ...elements.map((element) => <TableElement element={element} />),
                ];
            }
        } else if (props.isLoading === true) {
            return <Loader />;
        }
        return '';
    };

    return <p className={'table'}>{renderTable(props.elements, props.filteredElements)}</p>;
};

const mapStateToProps = (state) => {
    return {
        elements: state.data.elements,
        filteredElements: state.data.filteredElements,
        isLoading: state.appStatus.isLoading,
        currentElement: state.data.currentElement,
        sortingMap: state.filter.sortingMap,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(
            {
                setSelectedElement,
                setSortingMap,
                loadData,
            },
            dispatch
        ),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Table);
