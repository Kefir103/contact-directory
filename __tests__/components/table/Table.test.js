import React from 'react';
import { shallow, render } from 'enzyme';
import { initialState } from '../../../src/client/redux/initialState';
import configureMockStore from 'redux-mock-store';
import Table from '../../../src/client/components/table/Table';
import { getElementsByPage } from '../../../src/client/functions/dataFunctions';

const mockStore = configureMockStore();

describe('<Table />', () => {
    test('should render correctly with loaded fullData and pageCount = 2', () => {
        const element = {
            id: 1,
            firstName: 'firstName',
            lastName: 'lastName',
            email: 'email',
            phone: 'phone',
        };
        const pageCount = 2;
        const fullData = [];
        for (let i = 0; i < 100; i++) {
            fullData.push(element);
        }
        const currentElements = getElementsByPage(fullData, 1);
        const store = mockStore({
            ...initialState,
            data: {
                ...initialState.data,
                fullData,
                pageCount,
                currentElements,
            },
        });
        const wrapper = shallow(<Table store={store} />);
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
    test('should render correctly with loaded fullData and pageCount = 2', () => {
        const element = {
            id: 1,
            firstName: 'firstName',
            lastName: 'lastName',
            email: 'email',
            phone: 'phone',
        };
        const pageCount = 1;
        const fullData = [];
        for (let i = 0; i < 100; i++) {
            fullData.push(element);
        }
        const currentElements = getElementsByPage(fullData, 1);
        const store = mockStore({
            ...initialState,
            data: {
                ...initialState.data,
                fullData,
                pageCount,
                currentElements,
            },
        });
        const wrapper = shallow(<Table store={store} />);
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
    test('should render correctly without loaded fullData', () => {
        const fullData = [];
        const store = mockStore({
            ...initialState,
            data: {
                ...initialState.data,
                fullData,
            },
        });
        const wrapper = shallow(<Table store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
    test('should render correctly with filterText and empty appElements', () => {
        const element = {
            id: 1,
            firstName: 'firstName',
            lastName: 'lastName',
            email: 'email',
            phone: 'phone',
        };
        const fullData = [];
        for (let i = 0; i < 100; i++) {
            fullData.push(element);
        }
        const store = mockStore({
            ...initialState,
            filter: {
                ...initialState.filter,
                filterText: 'filterText',
            },
            data: {
                ...initialState.data,
                fullData,
            },
        });
        const wrapper = shallow(<Table store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
});
