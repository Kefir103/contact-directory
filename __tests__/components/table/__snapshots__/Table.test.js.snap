// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render correctly with filterText and empty appElements 1`] = `
<p
  id="filter-empty-message"
>
  Нет элементов списка, которые соответствуют Вашему запросу
</p>
`;

exports[`<Table /> should render correctly with loaded fullData and pageCount = 2 1`] = `
<Fragment>
  <div
    className="table"
    key="table"
  >
    <Connect(TablePanel) />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={0}
      key="tableElement0"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={1}
      key="tableElement1"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={2}
      key="tableElement2"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={3}
      key="tableElement3"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={4}
      key="tableElement4"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={5}
      key="tableElement5"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={6}
      key="tableElement6"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={7}
      key="tableElement7"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={8}
      key="tableElement8"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={9}
      key="tableElement9"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={10}
      key="tableElement10"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={11}
      key="tableElement11"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={12}
      key="tableElement12"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={13}
      key="tableElement13"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={14}
      key="tableElement14"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={15}
      key="tableElement15"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={16}
      key="tableElement16"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={17}
      key="tableElement17"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={18}
      key="tableElement18"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={19}
      key="tableElement19"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={20}
      key="tableElement20"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={21}
      key="tableElement21"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={22}
      key="tableElement22"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={23}
      key="tableElement23"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={24}
      key="tableElement24"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={25}
      key="tableElement25"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={26}
      key="tableElement26"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={27}
      key="tableElement27"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={28}
      key="tableElement28"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={29}
      key="tableElement29"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={30}
      key="tableElement30"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={31}
      key="tableElement31"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={32}
      key="tableElement32"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={33}
      key="tableElement33"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={34}
      key="tableElement34"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={35}
      key="tableElement35"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={36}
      key="tableElement36"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={37}
      key="tableElement37"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={38}
      key="tableElement38"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={39}
      key="tableElement39"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={40}
      key="tableElement40"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={41}
      key="tableElement41"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={42}
      key="tableElement42"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={43}
      key="tableElement43"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={44}
      key="tableElement44"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={45}
      key="tableElement45"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={46}
      key="tableElement46"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={47}
      key="tableElement47"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={48}
      key="tableElement48"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={49}
      key="tableElement49"
    />
  </div>
  <div
    className="paginator-container"
    key="paginatorContainer"
  >
    <Connect(Paginator)
      key="paginator"
    />
  </div>
</Fragment>
`;

exports[`<Table /> should render correctly with loaded fullData and pageCount = 2 2`] = `
<Fragment>
  <div
    className="table"
    key="table"
  >
    <Connect(TablePanel) />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={0}
      key="tableElement0"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={1}
      key="tableElement1"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={2}
      key="tableElement2"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={3}
      key="tableElement3"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={4}
      key="tableElement4"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={5}
      key="tableElement5"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={6}
      key="tableElement6"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={7}
      key="tableElement7"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={8}
      key="tableElement8"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={9}
      key="tableElement9"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={10}
      key="tableElement10"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={11}
      key="tableElement11"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={12}
      key="tableElement12"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={13}
      key="tableElement13"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={14}
      key="tableElement14"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={15}
      key="tableElement15"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={16}
      key="tableElement16"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={17}
      key="tableElement17"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={18}
      key="tableElement18"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={19}
      key="tableElement19"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={20}
      key="tableElement20"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={21}
      key="tableElement21"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={22}
      key="tableElement22"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={23}
      key="tableElement23"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={24}
      key="tableElement24"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={25}
      key="tableElement25"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={26}
      key="tableElement26"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={27}
      key="tableElement27"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={28}
      key="tableElement28"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={29}
      key="tableElement29"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={30}
      key="tableElement30"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={31}
      key="tableElement31"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={32}
      key="tableElement32"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={33}
      key="tableElement33"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={34}
      key="tableElement34"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={35}
      key="tableElement35"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={36}
      key="tableElement36"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={37}
      key="tableElement37"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={38}
      key="tableElement38"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={39}
      key="tableElement39"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={40}
      key="tableElement40"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={41}
      key="tableElement41"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={42}
      key="tableElement42"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={43}
      key="tableElement43"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={44}
      key="tableElement44"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={45}
      key="tableElement45"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={46}
      key="tableElement46"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={47}
      key="tableElement47"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={48}
      key="tableElement48"
    />
    <Connect(TableElement)
      element={
        Object {
          "email": "email",
          "firstName": "firstName",
          "id": 1,
          "lastName": "lastName",
          "phone": "phone",
        }
      }
      elementIndex={49}
      key="tableElement49"
    />
  </div>
  <div
    className="paginator-container"
    key="paginatorContainer"
  />
</Fragment>
`;

exports[`<Table /> should render correctly without loaded fullData 1`] = `null`;
